### PR TITLE
Support passing list of secrets to --preserve-env flag

### PR DESCRIPTION
**Pull request by Piccirello**
This PR changes the `--preserve-env` flag from a boolean flag to a string flag. It supports the same syntax as before (i.e. `--preserve-env` is equivalent to `--preserve-env=true`) while allowing you to specify a comma separated list of secrets to override. For example, `DEBUG=1 doppler run --preserve-env="DEBUG" -- printenv DEBUG`. To ensure backwards compatibility, I verified that the tests in the first commit passed before pushing up the second commit.

Closes ENG-6327.

### CODE REVIEW

File: pkg/cmd/run.go
Comment (watsonian) [inline pkg/cmd/run.go:697]: Is this a Cobra idiosyncrasy?

Comment (Piccirello) [reply pkg/cmd/run.go:697]: It is, afaik. Their [docs](https://github.com/spf13/pflag#command-line-flag-syntax) indicate that a space is only allowed on "non-boolean and flags without a 'no option default value'", and this flag now has a "no option default value".

Comment (watsonian) [reply pkg/cmd/run.go:697]: Well, that's annoying. Will this be reflected in the help output automatically? Or should we update the flag description?

Comment (erquhart) [reply pkg/cmd/run.go:697]: User that requested this here 👋 - I'd take having this critical functionality in it's current form over not having it. Possible to merge and iterate if/when necessary?

Comment (Piccirello) [reply pkg/cmd/run.go:697]: We're still waiting on reviews from the rest of the team but should have this shipped this week.

> Well, that's annoying. Will this be reflected in the help output automatically? Or should we update the flag description?

I'll add a note to the flag description.

**Review by nmanoogian — APPROVED** [review]

**Pull request merged**
